python/pyspark/errors/utils.py

```diff
@@ -197,6 +197,16 @@ def with_origin_to_class(cls: Type[T]) -> Type[T]:
     """
     if os.environ.get("PYSPARK_PIN_THREAD", "true").lower() == "true":
         for name, method in cls.__dict__.items():
-            if callable(method) and name != "__init__":
+            # Excluding Python magic methods that do not utilize JVM functions.
+            if callable(method) and name not in (
+                "__init__",
+                "__new__",
+                "__getattr__",
+                "__getitem__",
```

Comment:
I think `__getattr__` and `__getitem__` is still a PySpark expression? like `functions.getItem`

Reply:
Ah, yes we should remove `__getitem__` from the list but can we keep `__getattr__` here as it is?

Because `__getattr__` eventually calls `__getitem__` so the call site always overwritten by `__getitem__` in the end.

Reply:
If that's the case, I think we should list up all the aliases here?

Reply:
Seems like all the Column APIs directly calls the Spark API by itself other than `__getattr__`. But I think we can just remove it as well to keep consistency.